### PR TITLE
Initial puppet module for linux

### PR DESCRIPTION
**Pull request by jcheng-splunk**
Recreated PR and addressed comments from #374

- Ported from smart agent
- Install fluentd, plugins, and dependencies by default
- Use agent_config.yaml and fluent.conf from the collector package by default; can be configured to use custom files
- Add test job (lint, rake-spec, and pytest) to circleci

### CODE REVIEW

**Comment by jcheng-splunk** [comment]
@tmorrisonsfx PTAL again at the README.md.  I believe that I have addressed your feedback from #374.

**Comment by tmorrisonsfx** [comment]
> @tmorrisonsfx PTAL again at the README.md. I believe that I have addressed your feedback from #374.

@jcheng-splunk Yes, LGTM.

File: deployments/puppet/README.md [hunk 1]
```diff
@@ -0,0 +1,65 @@
...
+  splunk_realm => 'SPLUNK_REALM',
+}
+```
+
+This class accepts the following parameters:
+
+| Name | Description | Default value |
+| :--- | :---------- | :------------ |
+| `splunk_access_token` | **Required**: The Splunk access token to authenticate requests. | None |
+| `splunk_realm` | Which realm to send the data to.  The Splunk ingest and API URLs will be inferred by this value.  The `SPLUNK_REALM` environment variable will be set with this value for the collector service. | `us0` |
```
Comment (rmfitzpatrick) [inline deployments/puppet/README.md:34]: I think @jrcamp had the request in the ansible role to make this required w/o default

File: deployments/puppet/README.md [hunk 2]
```diff
@@ -0,0 +1,65 @@
...
+| `fluentd_config_source` | Source path to the fluentd config file.  This file will be copied to the `$fluentd_config_dest` path on the node.  See the [source attribute](https://puppet.com/docs/puppet/latest/types/file.html#file-attribute-source) of the `file` resource for supported value types.  The default source file is provided by the collector package. | `file:///etc/otel/collector/fluentd/fluent.conf` |
+| `fluentd_config_dest` | Destination path to the fluentd config file on the node. | `/etc/otel/collector/fluentd/fluent.conf` |
+| `manage_repo` | **Linux only**: In cases where the collector and fluentd apt/yum repositories are managed externally, set this to `false` to disable management of the repositories by this module.  **Note:** If set to `false`, the apt (`/etc/apt/sources.list.d/splunk-otel-collector.list`, `/etc/apt/sources.list.d/splunk-td-agent.list`) and yum (`/etc/yum.repos.d/splunk-otel-collector.repo`, `/etc/yum.repos.d/splunk-td-agent.repo`) repository definition files will be deleted if they exist in order to avoid any conflicts. | `true` |
```
Comment (rmfitzpatrick) [inline deployments/puppet/README.md:51]: Should we list the expected packages that their repos need to provide if this is false?

File: deployments/puppet/release [hunk 3]
```diff
@@ -0,0 +1,51 @@
...
+
+expected_output=$SCRIPT_DIR/pkg/signalfx-splunk_otel_collector-$(version_from_metadata).tar.gz
+if ! test -e $expected_output; then
+    echo "Module could not be built, expected output at $expected_output" >&2
+    exit 1
+fi
+
+echo "Module was successfully built and is available in ./pkg/$(basename $expected_output)"
+echo
+echo "You must manually upload this file in the Puppet forge web UI by following the directions at https://puppet.com/docs/puppet/5.4/modules_publishing.html#upload-a-module-to-the-forge"
```
Comment (rmfitzpatrick) [inline deployments/puppet/release:51]: not required here but something like these may be useful in future https://github.com/voxpupuli/puppet-blacksmith https://github.com/barnumbirr/action-forge-publish

Comment (jcheng-splunk) [reply deployments/puppet/release:51]: > not required here but something like these may be useful in future https://github.com/voxpupuli/puppet-blacksmith https://github.com/barnumbirr/action-forge-publish

Thanks!  I'll look into it.

File: deployments/puppet/spec/classes/init_spec.rb [hunk 4]
```diff
@@ -0,0 +1,24 @@
...
+  it "fails without access token" do
+    is_expected.to compile.and_raise_error(/splunk_access_token/)
+  end
+
+  on_supported_os.each do |os, facts|
+    if os.include? "windows"
+        next
+    end
+    context "on #{os}" do
+      let(:params) { { 'splunk_access_token' => "testing", } }
```
Comment (rmfitzpatrick) [inline deployments/puppet/spec/classes/init_spec.rb:16]: Think we need to populate the realm here now

**Review by rmfitzpatrick — APPROVED** [review]
lgtm w/ test updated

**Pull request merged**